pkgs/applications/networking/nextcloud-client/default.nix

```
@@ -20,13 +20,13 @@
 
 mkDerivation rec {
   pname = "nextcloud-client";
-  version = "3.1.3";
+  version = "3.2.0";
 
   src = fetchFromGitHub {
     owner = "nextcloud";
     repo = "desktop";
     rev = "v${version}";
-    sha256 = "sha256-8Ql6tOvWOjAvMJA87WlT9TbpnbciBsjDxRuYlMVi/m8=";
+    sha256 = "1nklsa2lx9ayjp8rk1mycjysqqmnq47djig0wygzna5mycl3ji06";
```

I think we should keep the `sha256-` prefix.

```suggestion
    sha256 = "sha256-1nklsa2lx9ayjp8rk1mycjysqqmnq47djig0wygzna5mycl3ji06";
```

---

1. How do you obtain a hash with that prefix?
2. I do not see that hash prefix being used much.
3. Why should you use one over the other?

---

I thought I had read somewhere that `sha256` is going to be replaced by `hash` at some point, which would make it necessary to specify the hash algorithm inside the hash string. However, I don't find that post anymore, so I'm really not sure.

@bryanasdev000 Do you remember why you added the `sha256-` prefix in 4515dfb6a23192d19b0eedd8564c888ce2474cfc?

---

Ah, you have already answered this question here: https://github.com/NixOS/nixpkgs/pull/113731/files#r580222460

> SRI hashes (https://www.w3.org/TR/SRI/) combine the hash algorithm and  a base-64 hash. This allows more concise and standard hash specifications.

https://github.com/NixOS/nix/commit/6024dc1d97212130c19d3ff5ce6b1d102837eee6

So I guess we could write:

```suggestion
    hash = "sha256-1nklsa2lx9ayjp8rk1mycjysqqmnq47djig0wygzna5mycl3ji06";
```

---

Looks like it's just a personal preference, not anything that has been decided yet, so I'll update my review accordingly.

---

Yeah, sorry about the confusion. The mention on that PR tricked my memory.

---

Just a note, if in any case the version is changed by a bot like ryantm's, the hash returns to the "new" model by default. In the end it doesn't make much difference but it is good to keep in mind.
Another thing, I use unstable Nix so by default it gives me this hash when I update git and I try to build it (my default way to know new hashes lol) I'm not sure if nix-prefetch- * is still using the old model.

---

I am using stable Nix and it gives me the non-SRI hashes.